providers/implementations/ciphers/cipher_chacha20_poly1305_hw.c

```diff
@@ -142,7 +142,7 @@ static int chacha20_poly1305_tls_cipher(PROV_CIPHER_CTX *bctx,
         ctx->len.text = plen;
 
         if (plen) {
-            if (ctx->enc)
+            if (bctx->enc)
```

Comment:
The other option is `ctx->base.enc`, which is the exact same thing, but is more in style with the rest of the function.

I leave it to you to decide.

Reply:
ctx->base.enc was my first attempt. Then I found bctx->... are used in multiple places and thought that seems cleaner.

Reply:
Okie